hal/tpm_io_linux.c

```diff
@@ -68,28 +68,34 @@
         #define TPM2_I2C_HZ   400000 /* 400kHz */
     #else
         /* SPI */
-        #ifdef WOLFTPM_MICROCHIP
-            /* Microchip ATTPM20 uses CE0 */
-            #define TPM2_SPI_DEV_CS "0"
-        #elif defined(WOLFTPM_ST33)
-            /* STM ST33HTPH SPI uses CE0 */
-            #define TPM2_SPI_DEV_CS "0"
-        #elif defined(WOLFTPM_NUVOTON)
-            /* Nuvoton NPCT75x uses CE0 */
-            #define TPM2_SPI_DEV_CS "0"
-        #else
-            /* OPTIGA SLB9670/SLB9762 and LetsTrust TPM use CE1 */
-            #define TPM2_SPI_DEV_CS "1"
+        #ifndef TPM2_SPI_DEV_CS
+            #ifdef WOLFTPM_MICROCHIP
+                /* Microchip ATTPM20 uses CE0 */
+                #define TPM2_SPI_DEV_CS "0"
+            #elif defined(WOLFTPM_ST33)
+                /* STM ST33HTPH SPI uses CE0 */
+                #define TPM2_SPI_DEV_CS "0"
+            #elif defined(WOLFTPM_NUVOTON)
+                /* Nuvoton NPCT75x uses CE0 */
+                #define TPM2_SPI_DEV_CS "0"
+            #else
+                /* OPTIGA SLB9670/SLB9762 and LetsTrust TPM use CE1 */
+                #define TPM2_SPI_DEV_CS "0"
+            #endif
+        #endif
+
+        #ifndef TPM2_SPI_DEV_PATH
+            #define TPM2_SPI_DEV_PATH "/dev/spidev0."
         #endif
 
         #ifdef WOLFTPM_AUTODETECT
             #undef TPM2_SPI_DEV
             /* this will try incrementing spidev chip selects */
-            static char TPM2_SPI_DEV[] = "/dev/spidev0.0";
+            static char TPM2_SPI_DEV[] = TPM2_SPI_DEV_PATH TPM2_SPI_DEV_CS;
```

Comment:
This auto-detect logic should always be 0.0 to 0.4.

Reply:
Got it.  I thought I was being smart by using the previously provided CS.  Much better to force this back to 0 to 4.  I'll update.